slowlog_functions.php

```diff
@@ -0,0 +1,1439 @@
+				'plugins/slowlog/slowlog.php' .
+				'?action=' . $tab_short_name .
+				(isset($_REQUEST['logid']) ? '&logid=' . $_REQUEST['logid']:'') .
+				(isset($_REQUEST['logentry']) ? '&logentry=' . $_REQUEST['logentry']:'')) .
+				"'>" . $tabs[$tab_short_name] . '</a></li>';
+
+			if (!isset($_REQUEST['logid'])) {
+				break;
+			}
+		}
+	}
+
+	print '</ul></nav></div>';
+}
+
+function get_table_associations($logid, $logentry = -1) {
+	load_reserved_words();
+
+	$sql = array();
+	$sql_prefix = 'INSERT INTO plugin_slowlog_details_tables (logid, logentry, table_name) VALUES ';
+	$sql_suffix = 'ON DUPLICATE KEY UPDATE table_name=VALUES(table_name)';
+
+	if ($logentry == -1) {
+		$rows = db_fetch_assoc_prepared('SELECT *
+			FROM plugin_slowlog_details
+			WHERE logid = ?',
+			array($logid));
+	} else {
+		$rows = db_fetch_assoc_prepared('SELECT *
+			FROM plugin_slowlog_details
+			WHERE logid = ? AND logentry = ?',
+			array($logid, $logentry));
+	}
+
+	foreach($rows as $row) {
+		$tokens = preg_split('/\s+/', $row['query']);
+
+		$query = $row['query'];
+		slowlog_debug('--------------------------------------------------------------------');
+		slowlog_debug(substr($row['query'],0,4000));
+
+		$in_delete   = false;
+		$in_show     = false;
+		$in_select   = false;
+		$in_from     = false;
+		$in_as       = false;
+		$in_join     = false;
+		$in_on       = false;
+		$in_using    = false;
+		$in_where    = false;
+		$in_having   = false;
+		$in_limit    = false;
+		$in_order    = false;
+		$in_groupby  = false;
+
+		$in_update   = false;
+		$in_table    = false;
+		$in_modify   = false;
+		$in_change   = false;
+		$in_set      = false;
+		$in_insert   = false;
+		$in_into     = false;
+		$in_values   = false;
+		$in_truncate = false;
+		$in_with     = false;
+
+		$table_mod   = false;
+
+		$tables = array();
+
+		foreach($tokens as $index => $t) {
+			$t = trim(strtolower($t));
+			$bparen = false;
+			$eparen = false;
+
+			if (substr($t, 0, 1) == '(') {
+				$bparen = true;
+				//slowlog_debug('Bparen: ' . $t);
+			} elseif (substr($t, 0, 1) == ')') {
+				$eparen = true;
+			}
+
+			$t = trim($t, ')(');
+
+			if ($t == '') {
+				continue;
+			}
+
+			//slowlog_debug('Token: ' . $t);
+
+			switch($t) {
+				case 'delete':
+					slowlog_debug('In DELETE');
+
+					$in_delete   = true;
+					$in_show     = false;
+					$in_select   = false;
+					$in_from     = false;
+					$in_as       = false;
+					$in_join     = false;
+					$in_on       = false;
+					$in_using    = false;
+					$in_where    = false;
+					$in_groupby  = false;
+					$in_having   = false;
+					$in_limit    = false;
+					$in_order    = false;
+
+					$in_update   = false;
+					$in_table    = false;
+					$in_modify   = false;
+					$in_change   = false;
+					$in_set      = false;
+					$in_insert   = false;
+					$in_into     = false;
+					$in_values   = false;
+					$in_truncate = false;
+					$in_with     = false;
```

Comment:
Set everything to false before the switch statement rather than repeating the code making it harder to read whats actually happening.

Reply:
This was a hack in general.  In the end, I went with the switch approach vs. having a set of functions to work their respective object types.  Do that in the next release.